src/pip/_internal/vcs/git.py

```diff
@@ -322,7 +322,20 @@ def get_remote_url(cls, location):
                 found_remote = remote
                 break
         url = found_remote.split(' ')[1]
-        return url.strip()
+        return cls._git_remote_to_pip_url(url.strip())
+
+    @staticmethod
+    def _git_remote_to_pip_url(url):
+        # type: (str) -> str
+        """
+        Convert a remote url from what git uses:
+            git@example.com:foo/bar.git
+        to what pip uses:
+            ssh://example.com/foo/bar.git
+
+        Leave https remotes alone.
```

Comment:
I think this leaves all URLs with scheme along (not just `https://` but also `ssh://`) so this is not accurate?

Reply:
Ohh, actually maybe hold off the review. Having read that git protocol page I realize that it's not just two distinct https or git cases. The shorthand syntax with the `:` before the project name can be used in either and is optional...